src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java

```diff
@@ -1,10 +1,160 @@
+				
+			} catch (NumberFormatException e) {
+				logger.log(Level.WARNING, "Invalid number for Student year", e);
+				continue;
+			}
+		}
+		
+		//Close scanner and return array of student IDs
+		scanner.close();
+		return formatStudentYears(studentIDs);
+		
+		} catch (NumberFormatException e) {
+			logger.log(Level.WARNING, "Invalid number for number of Students", e);
+			scanner.close();
+			return null;
+		}
+	}
+	
+	//Convert ArrayList<Integer> to int[]
+	private int[] formatStudentYears(ArrayList<Integer> studentIDs) {
+		int[] studentIDPrimitive = new int[studentIDs.size()];
+		
+		for (int i = 0; i < studentIDs.size(); i++) {
+			studentIDPrimitive[i] = studentIDs.get(i);
+		}
+		return studentIDPrimitive;
+		
+	}
+	
+	//Check for valid year and create Student from input
+	private Student checkYearAndCreateStudent(String stringYear, ArrayList<Integer> studentIDs, String firstName, String lastName) {
+		
+		int year = Integer.parseInt(stringYear);
+		
+		//Early return if year is not valid
+		if (year < 1 || year > 7) {
+			
+			logger.log(Level.WARNING, "Invalid year.");
+			return null;
+		}
+		
+		//Initialize boolean variable, to declare if new Student has been found in allStudents
+		boolean alreadyInList = false;
+		
+		//Loop through allStudents, checking to see if any existing entries have first name, last name and year from argument list
+		for (Student oneStudent : allStudents) {
+			
+			if (oneStudent.getFirstName().equals(firstName) &&
+				oneStudent.getLastName().equals(lastName) &&
+				oneStudent.getYear() == year) {
+				
+					//If so, set boolean to true and break the loop
+					alreadyInList = true;
+					logger.log(Level.WARNING, "Student already in database.");
+					break;
+			}
+		}
+		
+		//Create Student if Student not in list, else return null
+		if (!alreadyInList) {
+			
+				Student student = new Student(firstName, lastName, year);
+				
+				studentIDs.add(student.getStudentId());
+				
+				allStudents.add(student);
+				
+				logger.log(Level.INFO, "New Student created.");
+				return student;
+				
+			//Otherwise, return null	
+			} else {
+				
+				return null;
+				
+			}
+	}
+	//WTCET-19 - NEW until 154
+	//Show Student status - pass Student ID as an argument
+	public String showStatus(int studentId) {
+		
+		//Initialize String to receive Student details when found
+		String correctStudent = null;
+		
+		//Loop through list of Students, checking to see if the ID matches the one passed
+		for (Student student : allStudents) {
+			
+			//If so, assign to correctStudent and break the loop
+			if (student.getStudentId() == studentId) {
+				
+				correctStudent = student.toString();
+				break;
+			}
+		}
+		
+		//Log a warning if an invalid ID has been passed
+		if (correctStudent == null) logger.log(Level.WARNING, "ID does not exist.");
+		
+		//Return either the Student info or null, depending on the outcome of the loop
+		return correctStudent;
```

Comment:
Shouldn't it be logging instead of returning. I mean returning makes sense, but we should be logging as well I guess.

Reply:
No worries - I do need to return a value from this one, but can definitely add a confirmatory log here (and other places where something has gone successfully, now I think of it!) too.